added support for PICA shield and Z rack&pinion mechanism

mjrice opened the pull request:
added board id and pins definitions for the pica board.

thinkyhead committed:
RC7 README

thinkyhead committed:
Merge branch 'RCBugFix' into RC

thinkyhead committed:
README and Version for 1.1.0-RC8

thinkyhead committed:
Merge post-release patch from 'RCBugFix' into RC

robimarko committed:
Add missing string,fix lenght

robimarko committed:
Delete language_hr.h

robimarko committed:
Updated to add missing string and fix lenght

thinkyhead committed:
Merge pull request #5400 from robimarko/RC

Updated language_hr.h

thinkyhead committed:
README up-to-date

thinkyhead committed:
Patch Spanish language to fix redundant defines

LIII-XXII committed:
Update language_fr.h

Added missing accents for French locale. All added characters are present in latin1 extension.

thinkyhead committed:
Merge pull request #5690 from LIII-XXII/patch-1

Update language_fr.h

mjrice committed:
added support for PICA shield

added board id and pins definitions for the pica board.

Bob-the-Kuhn commented on Marlin/boards.h:
```diff
@@ -94,6 +94,8 @@
 #define BOARD_BAM_DICE          401  // 2PrintBeta BAM&DICE with STK drivers
 #define BOARD_BAM_DICE_DUE      402  // 2PrintBeta BAM&DICE Due with STK drivers
 #define BOARD_BQ_ZUM_MEGA_3D    503  // bq ZUM Mega 3D
+#define BOARD_PICA              184  // PICA Shield (rev C or later)
```
It's nice to be able to tell just by looking a the file names which one is newer.  I suggest using something like:
- pins_PICA_REVC.h & pins_PICA_REVB.h or
- pins_PICA.h & pins_PICA_OLDER.h

mjrice replied:
done!

Bob-the-Kuhn commented on Marlin/pins.h:
```diff
@@ -172,6 +172,8 @@
   #include "pins_MKS_13.h"
 #elif MB(SAINSMART_2IN1)
   #include "pins_SAINSMART_2IN1.h"
+#elif MB(PICA) || MB(PICA_REVB)
```
Usually there's one pins_xxxx.h for each motherboard.

In this case the pins_PICA_REVB.h would consist of your header and the following lines in this order:
```
#define BOARD_NAME "PICA_REVB" 
#include "pins_PICA.h"
```

You'll need to change the **# if MB(....** statement in pins_PICA.h to accommodate this change.

You may get a request to change the names of the boards so that it's easy to determine which is older just by looking at the names.  I suggest:
- pins_PICA_REVC.h & pins_PICA_REVB.h in case there should be a rev D in the future
- pins_PICA.h & pins_PICA_OLD.h style has also been used

I know you'll be able to find different naming conventions and the use of **# if MB(....** in other pins_XXXX.h files but it appears those have fallen out of favor.

thinkyhead replied:
Bob is correct.

thinkyhead committed:
Fix link to documentation (http)

We may drop this commit between releases.

mjrice committed:
split PICA pin definitions into two files corresponding to board revision

mjrice committed:
added support for rack & pinion Z probe (Z_RACK_PINION)

Added support for compiling with Z_RACK_PINION for machines like Wilson2
which have a Z probe mounted to a pinion that is deployed by the
movement of the X axis.
Changed the auto bedleveling default limits to be based on the bed size
instead of absolute values (configuration.h).

mjrice committed:
adding my configuration file (wilson 2 settings)

mjrice committed:
status screen displays "hit" next to each axis instead of location if the endstop is activated

Squid116 commented:
If we were to remove the z Rack and pinion stuff (everything after abae096) could we get PICA support up and running?
